components/service/glean/src/main/java/mozilla/components/service/glean/testing/GleanTestRule.kt

```diff
@@ -24,18 +24,27 @@ import org.junit.runner.Description
  * @get:Rule
  * val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
  * ```
+ *
+ * @param context the application context
+ * @param retainServerEndpoint whether or not the server endpoint should be retained
+ *        in the Glean SDK configuration across tests.
  */
 class GleanTestRule(
-    val context: Context
+    val context: Context,
+    val retainServerEndpoint: Boolean = false
 ) : TestWatcher() {
     override fun starting(description: Description?) {
         // We're using the WorkManager in a bunch of places, and Glean will crash
         // in tests without this line. Let's simply put it here.
         WorkManagerTestInitHelper.initializeTestWorkManager(context)
 
+        val config: Configuration = if (retainServerEndpoint && Glean.isInitialized()) {
```

Comment:
@mdboom @travis79 this looks like an hack to me, but I really couldn't find a better way around this. `Glean` is a singleton, so it gets reset by the test rule. However, the `GleanTestRunner` only initializes once. So we need to find a way for the test rule to not overwrite the url, if requested.

Reply:
Is there a way that you could just pass the `Configuration` object into the test rule? Maybe add a function that you could call?  I admit I don't know exactly what all is possible here so I need to research `Rules` a little more.

Reply:
That's a great idea! I did not think of that, it actually simplified things quite a lot. This now looks less like an hack. Thanks @travis79 !